pkg/resources/management.cattle.io/v3/setting/validator.go

```diff
@@ -69,71 +82,243 @@ func (v *Validator) Admitters() []admission.Admitter {
+	if s.Value == "" {
+		return nil
+	}
+
+	dur, err := validateDuration(s.Value)
+	if err != nil {
+		return field.TypeInvalid(valuePath, s.Value, err.Error())
+	}
+
+	// Note: zero duration is allowed and is equivalent to "".
+	if dur > 0 && a.isLessThanUserSessionTTL(dur) {
+		return field.Forbidden(valuePath, errLessThanAuthUserSessionTTL.Error())
+	}
+
+	return nil
+}
+
+// validateDeleteInactiveUserAfter validates the delete-inactive-user-after setting
+// to make sure it's a positive duration and that it's not less than the value of
+// auth-user-session-ttl-minutes setting and MinDeleteInactiveUserAfter.
+func (a *admitter) validateDeleteInactiveUserAfter(s *v3.Setting) error {
+	if s.Value == "" {
+		return nil
+	}
+
+	dur, err := validateDuration(s.Value)
+	if err != nil {
+		return field.TypeInvalid(valuePath, s.Value, err.Error())
+	}
+
+	// Note: zero duration is allowed and is equivalent to "".
+	if dur > 0 {
+		if dur < MinDeleteInactiveUserAfter {
+			err = fmt.Errorf("must be at least %s", MinDeleteInactiveUserAfter)
+		} else if a.isLessThanUserSessionTTL(dur) {
+			err = errLessThanAuthUserSessionTTL
 		}
-	default:
 	}
 
 	if err != nil {
-		return field.TypeInvalid(field.NewPath("value"), s.Value, err.Error())
+		return field.Forbidden(valuePath, err.Error())
 	}
 
 	return nil
 }
 
-func (a *admitter) validateAgentTLSMode(oldSetting, newSetting v3.Setting) error {
-	if oldSetting.Name != "agent-tls-mode" || newSetting.Name != "agent-tls-mode" {
+// validateUserRetentionCron validates the user-retention-cron setting
+// to make sure it's a valid standard cron expression.
+func (a *admitter) validateUserRetentionCron(s *v3.Setting) error {
+	if s.Value == "" {
 		return nil
 	}
+
+	if _, err := cron.ParseStandard(s.Value); err != nil {
+		return field.TypeInvalid(valuePath, s.Value, err.Error())
+	}
+
+	return nil
+}
+
+// validateUserLastLoginDefault validates the user-last-login-default setting
+// to make sure it's a valid RFC3339 formatted date time.
+func (a *admitter) validateUserLastLoginDefault(s *v3.Setting) error {
+	if s.Value == "" {
+		return nil
+	}
+
+	if _, err := time.Parse(time.RFC3339, s.Value); err != nil {
+		return field.TypeInvalid(valuePath, s.Value, err.Error())
+	}
+
+	return nil
+}
+
+// validateDuration parses the value as durations and makes sure it's not negative.
+func validateDuration(value string) (time.Duration, error) {
+	dur, err := time.ParseDuration(value)
+	if err != nil {
+		return 0, err
+	}
+
+	if dur < 0 {
+		return 0, errors.New("negative value")
+	}
+
+	return dur, err
+}
+
+// parseMinutes parses the value as minutes as returns the duration.
+func parseMinutes(value string) (time.Duration, error) {
+	minutes, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return time.Duration(minutes) * time.Minute, nil
+}
+
+func (a *admitter) validateAgentTLSMode(oldSetting, newSetting *v3.Setting) error {
```

Comment:
This change is deliberate to avoid unnecessarily dereferencing Setting objects.
The only case when `oldSetting` and `newSetting` can be `nil` is when `objectsv3.SettingOldAndNewFromRequest()` returns an err, which is checked.